components/aws/actions/s3-upload-file-tmp/s3-upload-file-tmp.mjs

```diff
@@ -1,38 +1,80 @@
 import common from "../../common/common-s3.mjs";
 import fs from "fs";
 import { toSingleLineString } from "../../common/utils.mjs";
+import { ConfigurationError } from "@pipedream/platform";
 
 export default {
   ...common,
   key: "aws-s3-upload-file-tmp",
   name: "S3 - Upload File - /tmp",
   description: toSingleLineString(`
-    Accepts a file path starting from /tmp, then uploads as a file to S3.
+    Accepts a file path or folder path starting from /tmp, then uploads the contents to S3.
     [See the docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/upload-objects.html)
   `),
-  version: "0.1.0",
+  version: "0.2.0",
   type: "action",
   props: {
     aws: common.props.aws,
     region: common.props.region,
     bucket: common.props.bucket,
-    filename: common.props.key,
+    filename: {
+      type: common.props.key.type,
+      label: common.props.key.label,
+      description: common.props.key.description,
+      optional: true,
+    },
     path: {
       type: "string",
       label: "File Path",
       description: "A path starting from `/tmp`, i.e. `/tmp/some_text_file.txt`",
+      optional: true,
+    },
+    folderPath: {
+      type: "string",
+      label: "Folder Path",
+      description: "A path starting from `/tmp`, i.e. `/tmp/some_folder`. If provided, all the files inside this path will be uploaded. This will override the `Filename Key` and `File Path` props.",
```

Comment:
```suggestion
A path starting from `/tmp`, for example, `/tmp/some_folder`. If provided, all the files inside this path will be uploaded. This will override the `Filename Key` and `File Path` props.
```